examples/threathunter/threat_intelligence/stix_parse.py

```diff
@@ -0,0 +1,348 @@
+from cybox.objects.domain_name_object import DomainName
+from cybox.objects.address_object import Address
+from cybox.objects.file_object import File
+from lxml import etree
+from io import BytesIO
+from stix.core import STIXPackage
+
+import logging
+import string
+import socket
+import uuid
+import time
+import datetime
+import dateutil
+import dateutil.tz
+
+from cabby.constants import (
+    CB_STIX_XML_111, CB_CAP_11, CB_SMIME,
+    CB_STIX_XML_10, CB_STIX_XML_101, CB_STIX_XML_11, CB_XENC_122002)
+
+CB_STIX_XML_12 = 'urn:stix.mitre.org:xml:1.2'
+
+BINDING_CHOICES = [CB_STIX_XML_111, CB_CAP_11, CB_SMIME, CB_STIX_XML_12,
+                   CB_STIX_XML_10, CB_STIX_XML_101, CB_STIX_XML_11,
+                   CB_XENC_122002]
+
+
+logger = logging.getLogger(__name__)
+
+#
+# Used by validate_domain_name function
+#
+domain_allowed_chars = string.printable[:-6]
+
+
+def validate_domain_name(domain_name):
+    """
+    Validate a domain name to ensure validity and saneness
+    :param domain_name: The domain name to check
+    :return: True or False
+    """
+    if len(domain_name) > 255:
+        logger.warn(
+            "Excessively long domain name {} in IOC list".format(domain_name))
+        return False
+
+    if not all([c in domain_allowed_chars for c in domain_name]):
+        logger.warn("Malformed domain name {} in IOC list".format(domain_name))
+        return False
+
+    parts = domain_name.split('.')
+    if 0 == len(parts):
+        logger.warn("Empty domain name found in IOC list")
+        return False
+
+    for part in parts:
+        if len(part) < 1 or len(part) > 63:
+            logger.warn("Invalid label length {} in domain name {} for report %s".format(
+                part, domain_name))
+            return False
+
+    return True
+
+
+def validate_md5sum(md5):
+    """
+    Validate md5sum
+    :param md5: md5sum to valiate
+    :return: True or False
+    """
+    if 32 != len(md5):
+        logger.warn("Invalid md5 length for md5 {}".format(md5))
+        return False
+    if not md5.isalnum():
+        logger.warn("Malformed md5 {} in IOC list".format(md5))
+        return False
+    for c in "ghijklmnopqrstuvwxyz":
+        if c in md5 or c.upper() in md5:
+            logger.warn("Malformed md5 {} in IOC list".format(md5))
+            return False
+
+    return True
+
+
+def sanitize_id(id):
+    """
+    Ids may only contain a-z, A-Z, 0-9, - and must have one character
+    :param id: the ID to be sanitized
+    :return: sanitized ID
+    """
+    return id.replace(':', '-')
+
+
+def validate_ip_address(ip_address):
+    try:
+        socket.inet_aton(ip_address)
+        return True
+    except socket.error:
+        return False
+
+
+def cybox_parse_observable(observable, indicator, timestamp, score):
```

Comment:
What is a cybox? Maybe a quick descriptive comment?

Reply:
I've added a short comment explaining cybox observables